packages/fxa-settings/src/pages/ResetPassword/ConfirmResetPassword/index.tsx

```diff
@@ -2,37 +2,74 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import React from 'react';
-import { RouteComponentProps } from '@reach/router';
-import { usePageViewEvent } from '../../../lib/metrics';
+import React, { useCallback, useState } from 'react';
+import { RouteComponentProps, useLocation, useNavigate } from '@reach/router';
+import Banner, { BannerType } from '../../../components/Banner';
 
 import LinkRememberPassword from '../../../components/LinkRememberPassword';
 import ConfirmWithLink, {
   ConfirmWithLinkPageStrings,
 } from '../../../components/ConfirmWithLink';
 import { REACT_ENTRYPOINT } from '../../../constants';
+import AppLayout from '../../../components/AppLayout';
+import { useAccount, useInterval } from '../../../models';
+import { FtlMsg } from 'fxa-react/dist/lib/utils';
+import { logPageViewEvent } from '../../../lib/metrics';
 
 export const viewName = 'confirm-reset-password';
 
 export type ConfirmResetPasswordProps = {
   email: string;
-  forceAuth?: boolean;
-  canSignIn?: boolean;
+  passwordForgotToken: string;
 };
 
-const ConfirmResetPassword = ({
-  email,
-  forceAuth,
-  canSignIn,
-}: ConfirmResetPasswordProps & RouteComponentProps) => {
-  usePageViewEvent(viewName, REACT_ENTRYPOINT);
+const POLLING_INTERVAL_MS = 2000;
 
-  // TODO check for passwordResetToken (confirms reset password initiated)
-  // TODO redirect to reset_password if !passwordResetToken
+const ConfirmResetPassword = (_: RouteComponentProps) => {
+  logPageViewEvent(viewName, REACT_ENTRYPOINT);
 
-  const resendHandler = () => {
-    // TODO resend code
-    // TODO logViewEvent metric
+  const navigate = useNavigate();
+  let { state = {} } = useLocation();
+  const account = useAccount();
+  const [passwordResetResend, setPasswordResetResend] = useState(false);
+  const [isPolling, setIsPolling] = useState<number | null>(
+    POLLING_INTERVAL_MS
+  );
+
+  const navigateToPasswordReset = useCallback(() => {
+    navigate('reset_password?showReactApp=true', { replace: true });
+  }, [navigate]);
+
+  if (!state) {
+    state = {};
+  }
+
+  let { email, passwordForgotToken } = state as ConfirmResetPasswordProps;
+
+  if (!email || !passwordForgotToken) {
+    navigateToPasswordReset();
+  }
+
+  useInterval(async () => {
```

Comment:
useInterval returns a callback function that can destroy the interval. Does this callback need need to be invoked at somepoint?